recaf-ui/src/main/java/me/coley/recaf/ui/control/NavigationBar.java

```diff
@@ -0,0 +1,173 @@
+package me.coley.recaf.ui.control;
+
+import javafx.geometry.Pos;
+import javafx.geometry.Side;
+import javafx.scene.Node;
+import javafx.scene.canvas.Canvas;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.control.*;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import javafx.scene.paint.Color;
+import me.coley.recaf.code.CommonClassInfo;
+import me.coley.recaf.code.FieldInfo;
+import me.coley.recaf.code.MemberInfo;
+import me.coley.recaf.code.MethodInfo;
+import me.coley.recaf.ui.CommonUX;
+import me.coley.recaf.ui.control.menu.ActionMenuItem;
+import me.coley.recaf.ui.util.Icons;
+import me.coley.recaf.util.logging.Logging;
+import org.slf4j.Logger;
+
+/**
+ * Navigation bar implementation. Helpful for easily navigating around a class.
+ * @author yapht
+ */
+public class NavigationBar extends HBox {
+    private static final Logger logger = Logging.get(NavigationBar.class);
+
+    private static class NavigationSeparator extends Canvas {
+        public NavigationSeparator() {
+            setWidth(5);
+            setHeight(15);
+
+            double pointY = (getHeight() / 2);
+            double pointX = getWidth() - 1;
+
+            // Draw a grey chevron
+            GraphicsContext gc = getGraphicsContext2D();
+            gc.setStroke(Color.rgb(90, 90, 90));
+            gc.strokeLine(0, 0, pointX, pointY);
+            gc.strokeLine(pointX, pointY, 0, getHeight());
+        }
+    }
+
+    private static class MemberNavigationNode extends Label {
+        private static class ScrollableContextMenu extends ContextMenu {
+            public ScrollableContextMenu() {
+                addEventHandler(Menu.ON_SHOWING, e -> {
+                    Node content = getSkin().getNode();
+                    if (content instanceof Region) {
+                        ((Region) content).setMaxHeight(getMaxHeight());
+                    }
+                });
+            }
+        }
+
+        private final CommonClassInfo classInfo;
+
+        public MemberNavigationNode(String text, CommonClassInfo classInfo) {
+            super(text);
+            this.classInfo = classInfo;
+
+            setOnMouseClicked(event -> onMouseClicked());
+            setGraphic(Icons.getClassIcon(classInfo));
```

Comment:
instead of `super(text)` can be `this(text, classInfo, Icons.getClassIcon(classInfo));` then remove the `this.classInfo = ...` and the other calls since its calling the other constructor.